src/coreComponents/codingUtilities/tests/testRTTypes.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2016-2024 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2024 Total, S.A
+ * Copyright (c) 2018-2024 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2023-2024 Chevron
+ * Copyright (c) 2019-     GEOS/GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#include "codingUtilities/RTTypes.hpp"
+#include "dataRepository/Group.hpp"
+#include "dataRepository/Wrapper.hpp"
+
+// TPL includes
+#include <gtest/gtest.h>
+#include <conduit.hpp>
+
+using namespace geos;
+using namespace dataRepository;
+
+// Mock classes to test dynamic casting
+class Base  {
+public:
+    virtual ~Base() = default; // Needed for RTTI
+};
+
+class Derived final : public Base {
+public:
+    void show() {
+        std::cout << "Derived class method." << std::endl;
+    }
+};
+
+// Test for dynamicCast with pointer
+TEST(DynamicCastTests, Pointer_Casting_Success) {
+    Base* base = new Derived();
+    Derived* derived = geos::dynamicCast<Derived*>(base);
+    ASSERT_NE(derived, nullptr) << "Expected successful cast from Base to Derived.";
+    delete base; // Clean up allocated memory
+}
+
+TEST(DynamicCastTests, Pointer_Casting_Failure) {
+    Base* base = new Base();
+    Derived* derived = geos::dynamicCast<Derived*>(base);
+    ASSERT_EQ(derived, nullptr) << "Expected nullptr due to failed cast from Base to Derived.";
+    delete base; // Clean up allocated memory
+}
+
+// Test for dynamicCast with reference
+TEST(DynamicCastTests, Reference_Casting_Success) {
+    Derived derived;
+    Base& base_ref = derived;
+    Derived& derived_ref = geos::dynamicCast<Derived&>(base_ref);
+    ASSERT_EQ(&derived_ref, &derived) << "Expected successful cast from Base to Derived.";
+}
+
+TEST(DynamicCastTests, Reference_Casting_Failure) {
+    Base base;
+    Base& base_ref = base;
+    
+    Base& derived_base_ref = geos::dynamicCast<Base&>(base_ref);
+    ASSERT_EQ(&derived_base_ref, &base) << "Expected successful cast from Base to Base.";
+    
+}
+
+
+// Typed test for geos wrapper
+template< typename T >
+class WrapperMock : public ::testing::Test
+{
+public:
+  WrapperMock():
+    m_node(),
+    m_group( "root", m_node ),
```

Comment:
do you need these 2?

Reply:
I guess the wrapper constructor nees a group... But what's the conduit node for?

Reply:
ah... you need to make a Group.. okay I was hoping we could avoid that include.